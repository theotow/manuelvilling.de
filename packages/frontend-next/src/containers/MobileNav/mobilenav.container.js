import React from 'react'
import { withRouter } from 'react-router'
import styled from 'styled-components'
import Text from '../../components/Text/text.component'
import Icon from '../../components/Icon/icon.component'
import { Ul, Li } from '../../components/DesktopNav/desktopnav.component'

const LiMobile = styled.li`
	box-sizing: border-box;
	padding-left: 15px;
	padding-right: 15px;
	height: 30px;
	border-bottom: 1px solid #000;
`

const UlMobile = styled.ul`
	display: block;
	position: absolute;
	left: 0;
	right: 0;
	margin-top: 12px;
	height: auto;
	background: white;
	border-top: 1px solid #000;
`

class MobileNav extends React.Component {
	state = {
		open: false,
	}

	toggle = () => {
		this.setState((state) => ({ open: !state.open }))
	}

	render() {
		return (
			<nav>
				<Ul>
					<Li>
						<a
							href="#"
							data-testid="mobile-nav-toggle-menu"
							onClick={this.toggle}>
							<Icon.Burger />
						</a>
					</Li>
				</Ul>
				{this.state.open && (
					<UlMobile>
						<LiMobile>
							<Text.ActiveStyledLinkMobile to="/">
								Home
							</Text.ActiveStyledLinkMobile>
						</LiMobile>
						<LiMobile>
							<Text.ActiveStyledLinkMobile to="/blog">
								Blog
							</Text.ActiveStyledLinkMobile>
						</LiMobile>
					</UlMobile>
				)}
			</nav>
		)
	}
}

const MobileNavWithRouter = withRouter(({ location }) => (
	<MobileNav key={location.pathname} />
))

export default MobileNavWithRouter
