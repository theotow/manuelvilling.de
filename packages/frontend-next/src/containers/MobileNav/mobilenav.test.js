import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import MobileNav from './mobilenav.container'
import { StaticRouter } from 'react-router-dom'

test('can open and close mobile menu', async () => {
	const { getByTestId, asFragment } = render(
		<StaticRouter location="/" context={{}}>
			<MobileNav />
		</StaticRouter>,
	)

	fireEvent.click(getByTestId('mobile-nav-toggle-menu'))

	expect(asFragment()).toMatchSnapshot() // open

	fireEvent.click(getByTestId('mobile-nav-toggle-menu'))

	expect(asFragment()).toMatchSnapshot() // closed
})
