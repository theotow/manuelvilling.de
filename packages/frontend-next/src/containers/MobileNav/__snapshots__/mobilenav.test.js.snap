// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can open and close mobile menu 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="desktopnavcomponent__Ul-sc-1ojer77-0 elKxFq"
    >
      <li
        class="desktopnavcomponent__Li-sc-1ojer77-1 ftjyxt"
      >
        <a
          data-testid="mobile-nav-toggle-menu"
          href="#"
        >
          <img
            alt="Burger"
            src="[object Object]"
          />
        </a>
      </li>
    </ul>
    <ul
      class="mobilenavcontainer__UlMobile-wupsm3-1 cUGEBJ"
    >
      <li
        class="mobilenavcontainer__LiMobile-wupsm3-0 cnhkrF"
      >
        <a
          class="textcomponent__StyledLink-sc-18da3xa-2 textcomponent__StyledLinkMobile-sc-18da3xa-3 yEZeS"
          data-active="true"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="mobilenavcontainer__LiMobile-wupsm3-0 cnhkrF"
      >
        <a
          class="textcomponent__StyledLink-sc-18da3xa-2 textcomponent__StyledLinkMobile-sc-18da3xa-3 eGqCuI"
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`can open and close mobile menu 2`] = `
<DocumentFragment>
  <nav>
    <ul
      class="desktopnavcomponent__Ul-sc-1ojer77-0 elKxFq"
    >
      <li
        class="desktopnavcomponent__Li-sc-1ojer77-1 ftjyxt"
      >
        <a
          data-testid="mobile-nav-toggle-menu"
          href="#"
        >
          <img
            alt="Burger"
            src="[object Object]"
          />
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
